app/controllers/api/databases.php

```diff
@@ -2564,22 +2799,78 @@ function updateAttribute(
-        }, $documents);
+        $documents = Authorization::skip(fn () => $dbForProject->find('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $queries));
+        $total = Authorization::skip(fn () => $dbForProject->count('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $filterQueries, APP_LIMIT_COUNT));
+
+        // Add $collectionId and $databaseId for all documents
+        $processDocument = function (Document $collection, Document $document) use (&$processDocument, $dbForProject, $database): bool {
+            $documentSecurity = $collection->getAttribute('documentSecurity', false);
+            $validator = new Authorization(Database::PERMISSION_READ);
+
+            $valid = $validator->isValid($collection->getRead());
+            if (!$documentSecurity && !$valid) {
+                return false;
+            }
+
+            $valid = $valid || $validator->isValid($document->getRead());
+            if ($documentSecurity && !$valid) {
+                return false;
+            }
+
+            $document->setAttribute('$databaseId', $database->getId());
+            $document->setAttribute('$collectionId', $collection->getId());
+
+            $relationships = \array_filter(
+                $collection->getAttribute('attributes', []),
+                fn ($attribute) => $attribute->getAttribute('type') === Database::VAR_RELATIONSHIP
+            );
+
+            foreach ($relationships as $relationship) {
+                $related = $document->getAttribute($relationship->getAttribute('key'));
+
+                if (empty($related)) {
+                    continue;
+                }
+                if (!\is_array($related)) {
+                    $relations = [$related];
+                } else {
+                    $relations = $related;
+                }
+
+                $relatedCollectionId = $relationship->getAttribute('relatedCollection');
+                $relatedCollection = Authorization::skip(fn() =>
+                    $dbForProject->getDocument('database_' . $database->getInternalId(), $relatedCollectionId));
+
+                foreach ($relations as $index => $doc) {
+                    if (!$processDocument($relatedCollection, $doc)) {
+                        unset($relations[$index]);
+                    }
+                }
+
+                if (\is_array($related)) {
+                    $document->setAttribute($relationship->getAttribute('key'), \array_values($relations));
+                } elseif (empty($relations)) {
+                    $document->setAttribute($relationship->getAttribute('key'), null);
+                }
+            }
+
+            return true;
+        };
+
+    // The linter is forcing this indentation
+    foreach ($documents as $index => $document) {
+        if (!$processDocument($collection, $document)) {
+            unset($documents[$index]);
+            $total--;
```

Comment:
It should still be correct, this only changes the total due to permissions. If you pass some queries, the initial total will be correct for those queries before permissions

Reply:
Let's say I limit to 10 but there are more documents. We only filter the 10 based on permissions, but not the others, right?

Reply:
Yep makes sense, should be fixed now, we can still use the database permission check for the `total` since it only ever checks the top level documents